### PR TITLE
Bumb Scalafmt version.

### PR DESCRIPTION
**Pull request by ckipp01**
Since we are going to merge in #1775, I figured this would be a good time to bump the version of scalafmt to 2.5.2 as well.

Wait until that pr is merged, and you should see the other two commits via @tgodzik disappear here, but I wanted to merge off that work for this.

### CODE REVIEW

File: .scalafmt.conf [hunk 1]
```diff
@@ -1,8 +1,7 @@
-version = "2.4.2"
+version = "2.5.2"
 project.git = true
-align = none
+align.preset=none
 docstrings = javadoc
-danglingParentheses = true
```
Comment (ckipp01) [inline .scalafmt.conf:5]: As far as i can tell, the multiple settings of `danglingParentheses` all default to `true`.

https://scalameta.org/scalafmt/docs/configuration.html#danglingparentheses

**Comment by ckipp01** [comment]
So there is a crazy amount of changes with this. Are there any addition settings that the rest of you want to add or take away with the bump to 2.5.x?

File: metals/src/main/scala/scala/meta/internal/builds/GradleBuildTool.scala [hunk 2]
```diff
@@ -17,27 +17,27 @@ case class GradleBuildTool(userConfig: () => UserConfiguration)
     val isSnapshotVersion = BuildInfo.gradleBloopVersion.contains("+")
     if (isSnapshotVersion)
       """|maven{
-         |  url 'https://dl.bintray.com/scalacenter/releases'
-         |}""".stripMargin
+        |  url 'https://dl.bintray.com/scalacenter/releases'
```
Comment (ckipp01) [inline metals/src/main/scala/scala/meta/internal/builds/GradleBuildTool.scala:20]: I really dislike the way it does these multiline strings in this way. Thoughts?

File: metals/src/main/scala/scala/meta/internal/builds/MillBuildTool.scala [hunk 3]
```diff
@@ -40,7 +40,9 @@ case class MillBuildTool(userConfig: () => UserConfiguration)
     // In some environments (such as WSL or cygwin), mill must be run using interactive mode (-i)
     val iOption = if (Properties.isWin) List("-i") else Nil
     val cmd =
-      iOption ::: "--predef" :: predefScriptPath(millVersion).toString :: "mill.contrib.Bloop/install" :: Nil
+      iOption ::: "--predef" :: predefScriptPath(
+        millVersion
```
Comment (tgodzik) [inline metals/src/main/scala/scala/meta/internal/builds/MillBuildTool.scala:44]: This is a bit weird, could split:
```scala
iOption ::: "--predef" :: predefScriptPath(millVersion).toString ::
 "mill.contrib.Bloop/install" :: Nil
```

File: metals/src/main/scala/scala/meta/internal/builds/SbtBuildTool.scala [hunk 4]
```diff
@@ -130,10 +130,10 @@ object SbtBuildTool {
       ""
     }
     s"""|// DO NOT EDIT! This file is auto-generated.
-        |// This file enables sbt-bloop to create bloop config files.
-        |$resolvers
-        |addSbtPlugin("ch.epfl.scala" % "sbt-bloop" % "$bloopSbtVersion")
-        |""".stripMargin
+       |// This file enables sbt-bloop to create bloop config files.
```
Comment (tgodzik) [inline metals/src/main/scala/scala/meta/internal/builds/SbtBuildTool.scala:133]: Should this be changed after adding the align setting?

Comment (ckipp01) [reply metals/src/main/scala/scala/meta/internal/builds/SbtBuildTool.scala:133]: Yea there still seems to be some formatting on multilines happening after the addition of the align setting. I'll look at this week and ensure everything is working fine and address the points above. Plus, some of the formatting on the multilines are also breaking tests.

**Comment by ckipp01** [comment]
Also just a note on this. The issue with interpolated string that I have mentioned [here](https://github.com/scalameta/scalafmt/issues/1998) is sort of a blocker on us doing this update. We have a lot of multiline strings that are being checked in tests etc, so this breaks quite a bit. I plan on waiting for that issue to addressed, and then I'll continue on with this.

**Comment by tgodzik** [comment]
@ckipp01 There is a new release, should we get back to it?

**Comment by ckipp01** [comment]
> @ckipp01 There is a new release, should we get back to it?

Yes, I'm actually going to close this and open another to not deal with the rebase.

**Pull request closed without merging**